src/redux/features/products/productThunks.tsx

```diff
@@ -0,0 +1,184 @@
+import { AxiosError } from "axios";
+import { RootState } from "../../app/store";
+import api from "../../../config/axios";
+
+const getProductsParams = (state: RootState) => {
+  const priceMin = state.filters.selectedFilters.price.min;
+  const priceMax = state.filters.selectedFilters.price.max;
+  const brands = state.filters.selectedFilters.brands;
+  const rating = state.filters.selectedFilters.rating;
+  const limit = state.product.limit;
+  const categoryId = state.filters.selectedFilters.category;
+  const [sortField, sortDirection] = state.product.sortValue.split("_");
+  const page = state.product.page;
+  const searchValue = state.filters.searchValue;
+
+  return {
+    page,
+    limit,
+    sortField,
+    sortDirection,
+    categoryId,
+    searchValue,
+    brands,
+    priceMin,
+    priceMax,
+    rating,
+  };
+};
+
+export const getProduct = createAsyncThunk<
+  Product,
+  GetProductPayload,
+  { rejectValue: string }
+>("products/product", async (payload, thunkAPI) => {
+  try {
+    const { _id } = payload;
+    const response = await api.get(`/products/${_id}`);
+
+    return response.data;
+  } catch (error) {
+    const axiosError = error as AxiosError;
+
+    return thunkAPI.rejectWithValue(axiosError.message);
+  }
+});
+
+export const getAllProducts = createAsyncThunk<
+  {
+    products: Product[];
+    total: number;
+    page: number;
+    limit: number;
+  },
+  void,
+  { rejectValue: string }
+>("product/getAllProducts", async (_, thunkAPI) => {
+  try {
+    const state: RootState = thunkAPI.getState() as RootState;
+    const { data } = await api.get("/products", {
+      params: {
+        ...getProductsParams(state),
+      },
+    });
+
+    return {
+      products: data.items,
+      total: data.total,
+      page: data.page,
+      limit: data.limit,
+    };
+  } catch (error) {
+    const axiosError = error as AxiosError;
+
+    return thunkAPI.rejectWithValue(axiosError.message);
+  }
+});
+
+export const showMoreProducts = createAsyncThunk<
+  {
+    products: Product[];
+    total: number;
+    page: number;
+    limit: number;
+  },
+  void,
+  { rejectValue: string }
+>("product/showMoreProducts", async (_, thunkAPI) => {
+  const state: RootState = thunkAPI.getState() as RootState;
+  const page = (state.product.showMorePage || state.product.page) + 1;
+
+  try {
+    const { data } = await api.get("/products", {
+      params: {
+        ...getProductsParams(state),
+        page,
+      },
+    });
+
+    return {
+      products: data.items,
+      total: data.total,
+      page: data.page,
+      limit: data.page,
+    };
+  } catch (error) {
+    const axiosError = error as AxiosError;
+
+    return thunkAPI.rejectWithValue(axiosError.message);
+  }
+});
+
+export const searchProducts = createAsyncThunk<
+  {
+    products: Product[];
+    total: number;
+    page: number;
+    limit: number;
+  },
+  { searchValue: string },
+  { rejectValue: string }
```

Comment:
Better to have types here

Reply:
@VYaniv , I already have type for it in one of the next PL